lib/pg.utils.js

```diff
@@ -30,18 +32,37 @@ const PREDEFINED_DOMAINS = {
   DateDay: 'date',
   DateTime: 'timestamp with time zone',
   Id: 'bigint',
-  JSON: 'json',
+  JSON: 'jsonb',
 };
 
 const IGNORED_DOMAINS = ['List', 'Enum', 'HashMap', 'HashSet'];
 
 const isValidIdentifier =
   name => !'0123456789'.includes(name[0]); // TODO: implement validation
 
+const isReserved =
+  name => RESERVED.includes(name.toUpperCase);
+
+const validateIdentifier = (name, type, prefix = '') => {
+  if (!isValidIdentifier(name)) {
+    throw new Error(
+      `Cannot create ${type} ${prefix}${name}` +
+      'because it is not a valid identifier'
+    );
+  }
+  if (isReserved(name)) {
+    throw new Error(
+      `Cannot create ${type} ${prefix}${name} because it is a reserved keyword`
+    );
+  }
+};
+
 module.exports = {
   escapeString,
   escapeValue,
   PREDEFINED_DOMAINS,
   IGNORED_DOMAINS,
   isValidIdentifier,
+  isReserved,
```

Comment:
This function is not used outside of this file, I am not sure if it should be exported.